packages/auth/src/authClients/supabase.ts

```diff
@@ -0,0 +1,32 @@
+import type { SupabaseAuthUser, SupabaseClient as Supabase, SupabaseAuthResponse} from '@supabase/supabase-js'
+
+import type { AuthClient } from './index'
+export type SupabaseUser = SupabaseAuthUser
+export type { Supabase }
+
+export interface AuthClientSupabase extends AuthClient {
+  login(options: {
+    email: string
+    password: string
+  }): Promise<SupabaseAuthResponse>
+  client: Supabase
+}
+
+export const supabase = (client: Supabase): AuthClientSupabase => {
+  return {
+    type: 'supabase',
+    client,
+    login: async ({ email, password }) =>
+      client.auth.login(email, password),
+    logout: async () => {
+      return client.auth.logout()
```

Comment:
Should 

`localStorage.getItem('supabase.auth.token')`

be cleared?

For example, in Supabase's "slack-clone" example, they remove the item and navigate to root:

```js
  signOut = () => {
    supabase.auth.logout()
    localStorage.removeItem('supabase-slack-clone')
    this.setState({ user: null })
    Router.push('/')
  }
```
 https://github.com/supabase/supabase/blob/master/examples/slack-clone/pages/_app.js#L39

You may also want to pass in options to logout to redirect to a "callback"-like page.

Reply:
Oh excellent! Supabase saves the token in a session:

https://github.com/supabase/supabase-js/blob/master/src/Auth.js#L59

```js
 if (this.persistSession) {
            const timeNow = Math.round(Date.now() / 1000)
            this.saveSession(
              this.accessToken,
              this.refreshToken,
              timeNow + tokenExpirySeconds,
              this.currentUser
            )
          }
```

So no need to mess with localStorage. Nice. Their Slack example was likely making it a little easier to fetch the user id.

Reply:
Yep! They remove it on logout too which is nice.